vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -667,6 +728,19 @@ def forward(self, hidden_states: torch.Tensor,
             activation=self.activation,
         )
 
+        if self.dp_size > 1:
+            if True:
+                all_hidden_states = get_dp_group().all_reduce(
```

Comment:
why do we need to have this allreduce?

Reply:
We need to reduce both within the DP group and within the TP group since each EP rank will have the partial accumulation of the output of the MoE layer. We could do reduce_scatter as well but I don't see a good way to do it without breaking CUDA graphs

Reply:
sounds good.